OpenRA.Mods.Common/Projectiles/Railgun.cs

```diff
@@ -145,17 +145,21 @@ public Railgun(ProjectileArgs args, RailgunInfo info, Color beamColor, Color hel
 			this.BeamColor = beamColor;
 			this.HelixColor = helixColor;
 
+			var world = args.SourceActor.World;
 			if (!string.IsNullOrEmpty(info.HitAnim))
-				hitanim = new Animation(args.SourceActor.World, info.HitAnim);
+			{
+				hitanim = new Animation(world, info.HitAnim);
+				world.ScreenMap.Add(this, target, hitanim.Image.Bounds);
+			}
 
-			CalculateVectors();
+			CalculateVectors(world);
 		}
 
-		void CalculateVectors()
+		void CalculateVectors(World world)
```

Comment:
`world` is already available from `args.SourceActor.World`, so why pass it in again here?